src/main/java/map/FileNavigator.java

```diff
@@ -0,0 +1,51 @@
+package map;
+
+import java.util.*;
+
+public class FileNavigator {
+    private final Map<String, List<FileData>> fileMap;
+
+    public FileNavigator() {
+        fileMap = new HashMap<>();
+    }
+
+    public void add(FileData fileData) {
+        String path = fileData.getPath();
+        List<FileData> files = fileMap.getOrDefault(path, new ArrayList<>());
+
+        for (FileData existingFile : files) {
+            if (!existingFile.getPath().equals(path)) {
+                System.out.println("There is an error because the key path and the file path do not match");
```

Comment:
You checking a little different condition. You check that your current map doesn't have files on the same path in FileData

Reply:
updated condition